cmd/bundle/remote_state.go

```diff
@@ -0,0 +1,73 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+	"github.com/databricks/cli/cmd/root"
+
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/spf13/cobra"
+)
+
+func newRemoteStateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remote-state",
+		Short: "Pull and print deployed state of the bundle",
```

Comment:
There could be situations when local state is newer than remote state (for example after `bundle run` command) and thus this command will read not remote but local state. So maybe we should chnage the name and desctiption of the command to just `state`?

Reply:
Hmm, can you explain how `bundle run` can result in this command reading the local state? The `remote-state` command by default will indeed not try to pull anything, but read the existing`.databricks/bundle/[target]/terraform/terraform.tfstate` file. Other commands can update it, but they only update it by pulling the terraform state from the workspace. By running the `remote-state` command without the `--force-pull` flag we might get outdated remote state from the local cache, but it will still be remote state (e.g. based on what was deployed as some point in the past). At least that's my understanding, but I might be missing something

Reply:
Some commands like bundle run, bundle deploy, bundle destroy execute terraform.Write as part of their execution 

So it means when there are local changes to bundle config and user runs “bundle run”, local state is updated. Bundle run does not push the state though. Not sure why we need to write state as part of run though cc @pietern

Reply:
Generally local state updated when bundle is being deployed, bundle config is transformed to TF state. If it’s the same or newer version then remote than it’s kept, otherwise newer remote version is used

Reply:
That local state is saved to a separate `.../bundle.tf.json`. I think it's needed for the `apply` operation, but not sure if `terraform.Load` looks into it, need to double check.

Reply:
@ilia-db ah yes, you're right, it only transforms to json not terraform state, so indeed it makes sense now. Then disregard my initial comment